File: 0012-integer-to-roman/0012-integer-to-roman.java
class Solution {
    public String intToRoman(int num) {
        int[] vals = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        String[] strs = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < 13; i++) {
            if (num == 0) {
                break;
            }
            if (vals[i] > num) {
                continue;
            }
            int times = num / vals[i];
            while (times > 0) {
                sb.append(strs[i]);
                times--;
            }
            num = num % vals[i];
        }
        
        return sb.toString();
    }
}

File: 0012-integer-to-roman/0012-integer-to-roman.cpp
class Solution {
public:
    string intToRoman(int num) {
        vector<string> sym {"I", "IV", "V", "IX", "X", "XL", "L", "XC", "C", "CD", "D", "CM", "M"};
        vector<int> val {1, 4, 5, 9, 10, 40, 50, 90, 100, 400, 500, 900, 1000};
        
        string res = "";
        
        for (int i = 12; i >= 0; i--) {
            if (num < 0) {
                break;
            }
            if (val[i] > num) {
                continue;
            }
            int times = num / val[i];
            while (times > 0) {
                res = res + sym[i];
                times--;
            }
            num = num % val[i];
        }
        
        return res;
    }
};